storybook/stories/Examples/Pie/CustomActiveShapePieChart.stories.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-shadow */
+import React, { useState } from 'react';
+import { Args } from '@storybook/react';
+import { Pie, PieChart, ResponsiveContainer, Sector } from '../../../../src';
+import { PieSectorDataItem } from '../../../../src/polar/Pie';
+
+export default {
+  component: Pie,
+};
+
+const data = [
+  { name: 'Group A', value: 400 },
+  { name: 'Group B', value: 300 },
+  { name: 'Group C', value: 300 },
+  { name: 'Group D', value: 200 },
+];
+
+export const CustomActiveShapePie = {
+  render: (args: Args) => {
+    const [activeIndex, setActiveIndex] = useState<number>(0);
+
+    const onPieEnter = (_: void, index: number) => {
+      setActiveIndex(index);
+    };
+
+    const renderActiveShape = (props: PieSectorDataItem) => {
+      const RADIAN = Math.PI / 180;
+      // @ts-expect-error property payload does not exist on type PieSectorDataItem
+      const { cx, cy, midAngle, innerRadius, outerRadius, startAngle, endAngle, fill, payload, percent, value } = props;
+      const sin = Math.sin(-RADIAN * midAngle);
+      const cos = Math.cos(-RADIAN * midAngle);
+      const sx = cx + (outerRadius + 10) * cos;
+      const sy = cy + (outerRadius + 10) * sin;
+      const mx = cx + (outerRadius + 30) * cos;
+      const my = cy + (outerRadius + 30) * sin;
+      const ex = mx + (cos >= 0 ? 1 : -1) * 22;
+      const ey = my;
+      const textAnchor = cos >= 0 ? 'start' : 'end';
+
+      return (
+        <g>
+          <text x={cx} y={cy} dy={8} textAnchor="middle" fill={fill}>
+            {payload.name}
+          </text>
+          <Sector
+            cx={cx}
+            cy={cy}
+            innerRadius={innerRadius}
+            outerRadius={outerRadius}
+            startAngle={startAngle}
+            endAngle={endAngle}
+            fill={fill}
+          />
+          <Sector
+            cx={cx}
+            cy={cy}
+            startAngle={startAngle}
+            endAngle={endAngle}
+            innerRadius={outerRadius + 6}
+            outerRadius={outerRadius + 10}
+            fill={fill}
+          />
+          <path d={`M${sx},${sy}L${mx},${my}L${ex},${ey}`} stroke={fill} fill="none" />
+          <circle cx={ex} cy={ey} r={2} fill={fill} stroke="none" />
+          <text x={ex + (cos >= 0 ? 1 : -1) * 12} y={ey} textAnchor={textAnchor} fill="#333">{`PV ${value}`}</text>
+          <text x={ex + (cos >= 0 ? 1 : -1) * 12} y={ey} dy={18} textAnchor={textAnchor} fill="#999">
```

Comment:
Optional: Can we simplify this example a bit more? Are all of these components needed to showcase the example?